ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/view/subviews/ToolVolumesFromListEditor.java

```diff
@@ -0,0 +1,204 @@
+package org.iplantc.de.admin.desktop.client.toolAdmin.view.subviews;
+
+import org.iplantc.de.admin.desktop.client.toolAdmin.ToolAdminView;
+import org.iplantc.de.admin.desktop.client.toolAdmin.model.ToolVolumesFromProperties;
+import org.iplantc.de.client.models.tool.ToolAutoBeanFactory;
+import org.iplantc.de.client.models.tool.ToolVolumesFrom;
+import org.iplantc.de.commons.client.validators.UrlValidator;
+
+import com.google.common.base.Strings;
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.editor.client.Editor;
+import com.google.gwt.editor.client.IsEditor;
+import com.google.gwt.event.logical.shared.ValueChangeEvent;
+import com.google.gwt.event.logical.shared.ValueChangeHandler;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.inject.Inject;
+import com.google.web.bindery.autobean.shared.AutoBean;
+import com.google.web.bindery.autobean.shared.AutoBeanUtils;
+
+import com.sencha.gxt.data.client.editor.ListStoreEditor;
+import com.sencha.gxt.data.shared.ListStore;
+import com.sencha.gxt.data.shared.ModelKeyProvider;
+import com.sencha.gxt.widget.core.client.Composite;
+import com.sencha.gxt.widget.core.client.event.CancelEditEvent;
+import com.sencha.gxt.widget.core.client.form.CheckBox;
+import com.sencha.gxt.widget.core.client.form.TextField;
+import com.sencha.gxt.widget.core.client.grid.ColumnConfig;
+import com.sencha.gxt.widget.core.client.grid.ColumnModel;
+import com.sencha.gxt.widget.core.client.grid.Grid;
+import com.sencha.gxt.widget.core.client.grid.editing.AbstractGridEditing;
+import com.sencha.gxt.widget.core.client.grid.editing.ClicksToEdit;
+import com.sencha.gxt.widget.core.client.grid.editing.GridEditing;
+import com.sencha.gxt.widget.core.client.grid.editing.GridRowEditing;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ToolVolumesFromListEditor extends Composite
+        implements IsEditor<Editor<List<ToolVolumesFrom>>> {
+
+    private static class StringValueChangeHandler implements ValueChangeHandler<String> {
+
+        TextField field;
+
+        public StringValueChangeHandler(TextField field) {
+            this.field = field;
+        }
+
+        @Override
+        public void onValueChange(ValueChangeEvent<String> event) {
+            String value = event.getValue();
+            if (value != null && value.isEmpty()) {
+                field.setValue(null);
+            }
+        }
+    }
```

Comment:
It would be useful to refactor these 3 duplicated private classes into a public class that could be used elsewhere (in metadata templates, for example). Maybe under `org.iplantc.de.commons.client.widgets`?